### PR TITLE
Update CONTRIBUTING.md guideline/manual

### PR DESCRIPTION
**Pull request by MarkusKuemmerle**

### CODE REVIEW

**Comment by MarkusKuemmerle** [comment]
See https://github.com/camaraproject/Governance/issues/59

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +48,26 @@ The following rule governs documentation contributions:
 
 * Contributions must be licensed under the [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0).
 
-At least if you add a new documentation to the repository, add your name into the [PARTICIPANTS](./PARTICIPANTS.MD) file.
+* At least if you add a new documentation to the repository, add your name into the [PARTICIPANTS](./PARTICIPANTS.MD) file or send an onboarding email to _project email address_.
```
Comment (hdamker) [inline CONTRIBUTING.md:51]: _project email address_ ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,7 +48,26 @@ The following rule governs documentation contributions:
...
+
+## Creating Branches
+
+Branches allow you to develop features, fix bugs, or safely experiment with new ideas in a contained area of our repository. If you need a friendly video introduction to git branching, please try [part two of "GitHub for Poets"](https://www.youtube.com/watch?v=oPpnCh7InLY) on Youtube, you may also read GitHub's ["Creating and deleting branches within your repository"](https://docs.github.com/en/articles/creating-and-deleting-branches-within-your-repository) for a deep dive.
```
Comment (hdamker) [inline CONTRIBUTING.md:55]: Contributor should normally a fork the repository before creating a branch (fork and pull model). Only codeowners have the right to create branches within the repository (?).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,27 +2,33 @@
 
 ## Code of conduct
 
-All participants of the project community must abide by the [CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md) derived from [Contributor Covenant](https://www.contributor-covenant.org/). Only by respecting each other can we develop a productive, collaborative community. Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting _project email address_ and/or a project maintainer.
+All participants of the CAMARA Project must abide by the [CODE_OF_CONDUCT.md](CODE_OF_CONDUCT.md) derived from [Contributor Covenant](https://www.contributor-covenant.org/). Only by respecting each other can we develop a productive, collaborative community. Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting _project email address_ and/or a Project maintainer.
```
[the remark this answers is not in the document]
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md:5]: ... is now inserted

File: CONTRIBUTING.md (hunk 1 above)
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: is now inserted

File: CONTRIBUTING.md (hunk 2 above)
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md:55]: For the moment I would like to keep it as simple as it is and don't mention the forking. If we get questions for clarification we can add it later.

Comment (hdamker) [reply CONTRIBUTING.md:55]: Most contributors have only view rights on the repositories, so they must use fork/pull model to contribute. And: already the third video of "Git & Github for poet" explains how this works.

Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: Working with forks added

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,27 +2,33 @@
...
+If you need a general basic and friendly introduction to git and GitHub, you are encouraged to watch the ["GitHub for poets"](https://www.youtube.com/watch?v=BCQHnlnPusY) series.
 
-* If you plan to do something more involved, please reach out to us and send an email to _project email address_. This will avoid unnecessary work and surely give you and us a good deal of inspiration.
+The starting point of any contribution should be an issue or a pull request:
 
-* Relevant coding style guidelines are available in the respective sub-repositories as they are programming language-dependent.
+* If you have a trivial fix or improvement, go ahead and create a pull request, addressing (with `@...`) a suitable maintainer of the respective repository (see the specific CODEOWNERS or MAINTAINERS.MD file of the Sub Project you want to contribute to) in the description of the pull request.
 
-## Steps to Contribute
+*	If you have a bigger change then please open an issue in the respective repository and put in all initial documents.
```
Comment (Kevsy) [inline CONTRIBUTING.md:19]: Suggest "If you have a bigger change that requires new document(s) to be contributed, then upload the initial doument(s) and open an issue in the respective repository. Make sure the issue includes links to the relevant document(s)."

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -42,32 +48,43 @@ The following rule governs documentation contributions:
...
+*	[Propose changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests)
+*	[Collaborate with pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests)
+
+We kindly ask you to follow the [Pull Request Checklist](#Pull-Request-Checklist) to ensure reviews can happen accordingly.
 
 ## Pull Request Checklist
 
+* Work on a fork of the repository.
+
 * Branch from the main branch and, if needed, rebase to the current main branch before submitting your pull request. If it doesn't merge cleanly with main you may be asked to rebase your changes.
```
Comment (Kevsy) [inline CONTRIBUTING.md:78]: Suggest add: "(see this [Git rebase tutorial for help](https://www.codementor.io/git/tutorial/git-rebase-split-old-commit-master)"

**Review by Kevsy — CHANGES_REQUESTED** [review]
Looks good, just some minor suggestions added inline :)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -31,8 +37,8 @@ You are welcome to contribute code in order to fix a bug or to implement a new f
 The following rule governs code contributions:
 
 * Contributions must be licensed under the [Apache 2.0 License](https://www.apache.org/licenses/LICENSE-2.0)
-* Newly created files must be opened by an instantiated version to the file 'templates/file-header.txt'
-* At least if you add a new file to the repository, add your name into the [PARTICIPANTS](./PARTICIPANTS.MD) file.
+* Newly created files must be opened by an instantiated version to the [file-header.txt](https://github.com/camaraproject/Governance/blob/main/declarations/templates/file-header.txt).
```
[the remark this answers is not in the document]
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md:40]: It means that each new file should start with the text in file-header.txt

File: CONTRIBUTING.md (hunk 4 above)
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: Adapted the text in your sense. But we would like to have the initial documents attached to the issues and not in the repository itself.

File: CONTRIBUTING.md (hunk 5 above)
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: Is added now

File: CONTRIBUTING.md (hunk 4 above)
Comment (Kevsy) [reply CONTRIBUTING.md]: OK understood.

File: CONTRIBUTING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kevsy) [reply CONTRIBUTING.md:40]: Ah, that makes sense  - can you use the text in your comment above text instead?

**Review by Kevsy — CHANGES_REQUESTED** [review]
just one change (see comment)

File: CONTRIBUTING.md (hunk 5 above)
Comment (hdamker) [reply CONTRIBUTING.md]: @Kevsy @MarkusKuemmerle My impression is that the "Git rebase tutorial" above is about a different specific use case of rebase (= rewriting the history of a branch, e.g. to spilt commits). I would recommend instead to point to the GitHub manual page [Keeping your pull request in sync with the base branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch) and maybe in addition [More ways to keep your pull request branch up-to-date](https://github.blog/changelog/2022-02-03-more-ways-to-keep-your-pull-request-branch-up-to-date/).

**Comment by hdamker** [comment]
Looks good to me as well, only just one comment (see https://github.com/camaraproject/Governance/pull/66#discussion_r992587954)

File: ProjectStructureAndRoles.md [hunk 7]
```diff
@@ -51,6 +51,8 @@ Now they can work on the solution and add commits to the branch. If they are don
 6 Finally one of the Codeowners of the Sub Project shall merge the pull request into the main branch. By that the Codeowner closes the pull request. The Codeowner also shall close the branch and the issue (if not already automatically done).
```
Comment (shilpa-padgaonkar) [inline ProjectStructureAndRoles.md:51]: A pull request review in certain cases might lead to a PR not being approved. In this case, the code owner shall close the pull request with the appropriate comments and the contribution shall not get merged into the main branch.

**Review by shilpa-padgaonkar — APPROVED** [review]
Just a minor comment https://github.com/camaraproject/Governance/pull/66/files#r992606647
Rest LGTM

File: CONTRIBUTING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: Have changed it

File: CONTRIBUTING.md (hunk 5 above)
Comment (MarkusKuemmerle) [reply CONTRIBUTING.md]: New links are included

File: ProjectStructureAndRoles.md (hunk 7 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:51]: Is added now

**Pull request merged**